packages/store/test/StoreCore.t.sol

```diff
@@ -838,7 +839,18 @@ contract StoreCoreTest is Test, StoreMock {
       new string[](1)
     );
 
-    IStore(this).registerStoreHook(table, subscriber);
+    IStore(this).registerStoreHook(
+      table,
+      subscriber,
+      EnabledStoreHooks({
+        onBeforeSetRecord: true,
+        onAfterSetRecord: false,
+        onBeforeSetField: true,
+        onAfterSetField: false,
+        onBeforeDeleteRecord: true,
+        onAfterDeleteRecord: false
+      })
```

Comment:
I wonder if this would also work
```solidity
EnabledStoreHooks({
  onBeforeSetRecord: true,
  onBeforeSetField: true,
  onBeforeDeleteRecord: true
})
```
(only toggling on the ones we want and letting the others default, but not sure if solc will complain)

Reply:
I also wonder if we should just use a `unit8` here instead of the struct so that the ABI (and presumably abi-encoded calldata) is a bit more compact vs `(bool,bool,bool,bool,...)`

Reply:
unfortunately `solc` does complain :(

Reply:
> I also wonder if we should just use a unit8 here

yeah i think you're right here - we just have to document how to use the helper libraries to encode the bitmap, but it would be much cheaper to encode it in eg. the module the hook is registered and only pass the `uint8` between contracts.